--- a/helix-term/src/ui/prompt.rs
+++ b/helix-term/src/ui/prompt.rs
@@ -604,6 +604,9 @@ fn handle_event(&mut self, event: &Event, cx: &mut Context) -> EventResult {
                     // handle executing with last command in history if nothing entered
                     let input = if self.line.is_empty() {
                         &last_item
+                    } else if self.line.starts_with("harpoon-open") {
+                        self.line = self.line.replace("harpoon-open", "open");
+                        &self.line
                     } else {
                         if last_item != self.line {
                             // store in history
